File: code-coaches/medium/flowing-words/flowing_words.rb
def does_sentence_flow(sentence)
  words = sentence.split
  (1...words.length).all? { |i| words[i][0] == words[i - 1][-1] }
end

def main
  sentence = gets.strip
  puts does_sentence_flow(sentence)
end

if __FILE__ == $0
  main
end

File: code-coaches/medium/flowing-words/flowing_words.py
def does_sentence_flow(sentence: str) -> bool:
    words = sentence.split()
    return all(words[i][0] == words[i - 1][-1] for i in range(1, len(words)))


def main():
    sentence = input()

    if does_sentence_flow(sentence):
        print("true")
    else:
        print("false")


if __name__ == "__main__":
    main()
